applications/RomApplication/python_scripts/rom_solver.py

```diff
@@ -0,0 +1,58 @@
+# Importing the Kratos Library
+import KratosMultiphysics
+
+# Import applications
+import KratosMultiphysics.RomApplication as KratosROM
+
+def CreateSolver(cls, model, custom_settings):
+    class ROMSolver(cls):
+        """ROM solver generic main class.
+        This class serves as a generic class to make a standard Kratos solver ROM-compatible.
+        It extends the default parameters to include the \'rom_settings\' and overrides the
+        creation of the builder and solver to use the ROM one.
+        """
+
+        def __init__(self, model, custom_settings):
+            super().__init__(model, custom_settings)
+            KratosMultiphysics.Logger.PrintInfo("::[ROMSolver]:: ", "Construction finished")
+
+        @classmethod
+        def GetDefaultParameters(cls):
+            default_settings = KratosMultiphysics.Parameters("""{
+                "rom_settings": {
+                    "nodal_unknowns": [],
+                    "number_of_rom_dofs": 0
+                }
+            }""")
+            default_settings.AddMissingParameters(super().GetDefaultParameters())
+            return default_settings
+
+        def _CreateBuilderAndSolver(self):
+            linear_solver = self._GetLinearSolver()
+            rom_parameters = self._ValidateAndReturnRomParameters()
+            builder_and_solver = KratosROM.ROMBuilderAndSolver(linear_solver, rom_parameters)
+            return builder_and_solver
+
+        def _ValidateAndReturnRomParameters(self):
+            # Check that the number of ROM DOFs has been provided
+            n_rom_dofs = self.settings["rom_settings"]["number_of_rom_dofs"].GetInt()
+            if not n_rom_dofs > 0:
+                err_msg = "\'number_of_rom_dofs\' in \'rom_settings\' is {}. Please set a larger than zero value.".format(n_rom_dofs)
+                raise Exception(err_msg)
+
+            # Check if the nodal unknowns have been provided by the user
+            # If not, take the DOFs list from the base solver
+            nodal_unknowns = self.settings["rom_settings"]["nodal_unknowns"].GetStringArray()
+            if len(nodal_unknowns) == 0:
+                solver_dofs_list = self.GetDofsList()
+                if not len(solver_dofs_list) == 0:
+                    self.settings["rom_settings"]["nodal_unknowns"].SetStringArray()
```

Comment:
```suggestion
                    self.settings["rom_settings"]["nodal_unknowns"].SetStringArray(solver_dofs_list)
```

Reply:
I understand this line should tackle the rom_settings.nodal_unknowns being empty.

However, I have porpusely erased it `{
    "rom_settings": {
        "nodal_unknowns": [], ...`

But doing this change alone will not work if the user indeed enters an empty list. 

Line 56 in rom_analisys.py should also be changed

Reply:
Nice catch! Indeed, while updating one of the examples in the Examples repo I realized that we need to slightly change the factory method in `rom_analysis.py`. I'd change in there.